app/controllers/api/v1/icons_controller.rb

```diff
@@ -50,9 +50,9 @@ def find_icon(ids)
         if ids[:icon_id].present?
           Icon.find(ids[:icon_id])
         elsif ids[:portfolio_item_id].present?
-          PortfolioItem.find(ids[:portfolio_item_id]).icon
+          Icon.find_by!(:restore_to => PortfolioItem.find(ids[:portfolio_item_id]))
         elsif ids[:portfolio_id].present?
-          Portfolio.find(ids[:portfolio_id]).icon
+          Icon.find_by!(:restore_to => Portfolio.find(ids[:portfolio_id]))
```

Comment:
Possibly I'm misunderstanding the `restore_to` url. Doesn't that return a url to the original icon?

Reply:
No, that is just the polymorphic designation so the `Icon.find_by!` will fail if there is no icon for the portfolio/portfolio_item specified.